src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.c

```diff
@@ -27,14 +28,42 @@ const ActorInit Obj_Dowsing_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+s32 ObjDowsing_GetFlag(ObjDowsing* this, GlobalContext* globalCtx) {
+    s32 type = DOWSING_GET_TYPE(this);
+    s32 flag = DOWSING_GET_FLAG(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dowsing/func_80B23D50.s")
+    if (type == DOWSING_COLLECTIBLE) {
+        return Actor_GetCollectibleFlag(globalCtx, flag);
+    } else if (type == DOWSING_CHEST) {
+        return Actor_GetChestFlag(globalCtx, flag);
+    } else if (type == DOWSING_SWITCH) {
+        return Flags_GetSwitch(globalCtx, flag);
+    } else {
+        return 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dowsing/func_80B23DD0.s")
+s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx) {
+    if (ObjDowsing_GetFlag(this, globalCtx)) {
+        Actor_MarkForDeath(&this->actor);
+        return 1;
+    }
+    return 0;
+}
```

Comment:
```suggestion
s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx) {
    if (ObjDowsing_GetFlag(this, globalCtx)) {
        Actor_MarkForDeath(&this->actor);
        return true;
    }
    return false;
}
```
This is used as a boolean. Can also probably call it something like `CheckValidSpawn`